purchase_order_type_advanced/models/purchase_order.py

```diff
@@ -0,0 +1,25 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.onchange("order_type")
+    def onchange_order_type(self):
+        super().onchange_order_type()
+        for order in self:
+            if order.order_type.picking_type_id:
+                order.picking_type_id = order.order_type.picking_type_id
+
+    def action_view_invoice(self):
+        res = super(PurchaseOrder, self).action_view_invoice()
+        ctx = {}
+        if self.order_type.journal_id:
+            ctx["default_journal_id"] = self.order_type.journal_id.id
+        if self.order_type:
+            ctx["default_purchase_type_id"] = self.order_type.id
+        res["context"].update(ctx)
```

Comment:
In v14 this `res["context"]` seems to be a string, not a dictionary, so this `update()` fails.
Tested in AR Test Environment, clicking in purchase invoices button, this error is raised:

```
  File "/home/odoo/odoo-test/erp-ariveira/src/slv-purchase/purchase_order_type_advanced/models/purchase_order.py", line 24, in action_view_invoice
    res["context"].update(ctx)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/odoo/odoo-test/erp-ariveira/src/odoo/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/odoo-test/erp-ariveira/src/odoo/odoo/http.py", line 316, in _handle_exception
    raise exception.with_traceback(None) from new_cause
AttributeError: 'str' object has no attribute 'update'
```

Reply:
Solved!